pkg/defaults/mesh.go

```diff
@@ -0,0 +1,48 @@
+package defaults
+
+import (
+	"context"
+
+	mesh_core "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
+)
+
+var defaultMeshKey = core_model.ResourceKey{
+	Mesh: core_model.DefaultMesh,
+	Name: core_model.DefaultMesh,
+}
+
+func (d *defaultsComponent) meshExists() (bool, error) {
+	mesh := &mesh_core.MeshResource{}
+	if err := d.resManager.Get(context.Background(), mesh, core_store.GetBy(defaultMeshKey)); err != nil {
+		if core_store.IsResourceNotFound(err) {
+			return false, nil
+		} else {
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+func (d *defaultsComponent) createDefaultMesh() error {
+	return d.resManager.Create(context.Background(), &mesh_core.MeshResource{}, core_store.CreateBy(defaultMeshKey))
+}
+
+func (d *defaultsComponent) createMeshIfNotExist() error {
+	exists, err := d.meshExists()
+	if err != nil {
+		return err
+	}
+	if exists {
+		log.V(1).Info("default Mesh already exists. Skip creating default Mesh.")
+	} else {
+		log.Info("trying to create default Mesh")
+		if err := d.createDefaultMesh(); err != nil {
+			log.V(1).Info("could not create default mesh", "err", err)
+			return err
+		}
+		log.Info("default Mesh created")
+	}
+	return nil
+}
```

Comment:
I believe `meshExists` and `createDefaultMesh` methods are excessive and also could be easily inlined and at the end, we will have less code
```suggestion
func (d *defaultsComponent) createMeshIfNotExist() error {
	mesh := &mesh_core.MeshResource{}
	err := d.resManager.Get(context.Background(), mesh, core_store.GetBy(defaultMeshKey)); 
	if err == nil {
		log.V(1).Info("default Mesh already exists. Skip creating default Mesh.")
		return nil
	}
	if !core_store.IsResourceNotFound(err) {
		return false, err
	}
	log.Info("trying to create default Mesh")
	if err := d.resManager.Create(context.Background(), &mesh_core.MeshResource{}, core_store.CreateBy(defaultMeshKey)); err != nil {
		log.V(1).Info("could not create default mesh", "err", err)
		return err
	}
	log.Info("default Mesh created")
	return nil
}
```

Reply:
Less code does not always means code that is easier to understand.

In this case, I'd argue that the first one is easier to understand (even with extra indentation which is anti-SOLID etc.). But I don't have a strong preference, hence I'm changing it